data-prepper-core/src/main/java/org/opensearch/dataprepper/plugin/ExtensionPluginConfigurationResolver.java

```diff
@@ -0,0 +1,29 @@
+package org.opensearch.dataprepper.plugin;
+
+import org.opensearch.dataprepper.model.configuration.PipelinesDataFlowModel;
+import org.opensearch.dataprepper.parser.model.DataPrepperConfiguration;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+@Named
+public class ExtensionPluginConfigurationResolver {
+    private final Map<String, Object> extensionMap;
+
+    @Inject
+    public ExtensionPluginConfigurationResolver(final DataPrepperConfiguration dataPrepperConfiguration,
```

Comment:
As noted in another comment, we should not tie extensions configurations directly with pipeline configurations. Can we decouple these somewhat?

Reply:
Would you mind if I do it in the other PR: https://github.com/opensearch-project/data-prepper/pull/3340. There has been some tweaks on the interface

Reply:
That seems reasonable.